### PR TITLE
[lldb] add de-duplication and working dir support for fmodule-file

### PR DESCRIPTION
**Pull request by rmaz**
This PR adds correct handling of the `-fmodule-file=` flag when
deserializing Swift debugging info. This flag is of the form 
`-fmodule-file=[<name>=]<file>`, so we need to split on the final 
`=` to get the path component in all cases.

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -1407,19 +1407,29 @@ bool ConsumeIncludeOption(StringRef &arg, StringRef &prefix) {
...
   for (StringRef &option : options)
     if (arg.consume_front(option)) {
       prefix = option;
       return true;
     }
+
+  // We need special handling for -fmodule-file as we need to
+  // split on the final = after the module name.
+  if (arg.startswith("-fmodule-file=")) {
+    prefix = arg.substr(0, arg.rfind("=") + 1);
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:1420]: I assume this is safe, even if arg is either "foo",  or "foo="?

File: lldb/unittests/Symbol/TestSwiftASTContext.cpp [hunk 2]
```diff
@@ -268,19 +268,82 @@ TEST_F(TestSwiftASTContext, ApplyWorkingDir) {
   EXPECT_EQ(
       single_arg_rel_path,
       llvm::SmallString<128>("-fmodule-map-file=rel/dir/module.modulemap"));
+
+  // fmodule-file needs to handle different cases:
+  //  -fmodule-file=path/to/pcm
+  //  -fmodule-file=name=path/to/pcm
```
Comment (adrian-prantl) [inline lldb/unittests/Symbol/TestSwiftASTContext.cpp:274]: here it says `-fmodule-file=name=path/to/pcm` but all the tests use `-fmodule-file=modulename=path/to/pcm` is that a tricky test for the edge-case of passing in something that ends with `name=` or a typo?

Comment (adrian-prantl) [reply lldb/unittests/Symbol/TestSwiftASTContext.cpp:274]: Thanks!

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 1 above)
Comment (rmaz) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:1420]: The two valid cases are `-fmodule-file=some/path` and `-fmodule-file=modulename=some/path`. We know the first `=` is present as we checked the prefix above, so this will split correctly in both cases. There is no form `-fmodule-file some/path` so we don't need to handle that case.

File: lldb/unittests/Symbol/TestSwiftASTContext.cpp (hunk 2 above)
Comment (rmaz) [reply lldb/unittests/Symbol/TestSwiftASTContext.cpp:274]: The `name` string can be anything, it is used to indicate the name of the module that you are providing the path to. I can make these the same if you think it is clearer? It shouldn't make any difference to the flag splitting logic though.

Comment (adrian-prantl) [reply lldb/unittests/Symbol/TestSwiftASTContext.cpp:274]: Oh, this isn't a key=value pair whre the key is "name", it's [name of the module]=[file name]!

**Review by adrian-prantl — APPROVED** [review]

File: lldb/unittests/Symbol/TestSwiftASTContext.cpp (hunk 2 above)
Comment (rmaz) [reply lldb/unittests/Symbol/TestSwiftASTContext.cpp:274]: Exactly, yes. I believe the idea is that the compiler can avoid loading pcm files until they are imported.
